src/Data/provideLocalStorageDataClass.ts

```diff
@@ -142,6 +145,34 @@ export function provideLocalStorageDataClass(
   }
 }
 
+export function searchLocalStorageDataClasses(
+  search: string,
+  blackListEntities: string[] = [],
+): Array<{ _id: string; entity: string; title: string }> {
+  const results: Array<{ _id: string; entity: string; title: string }> = []
+
+  const lowerCaseSearchTerm = search.toLowerCase()
+  const matchesSearchTerm = (value: unknown) =>
+    typeof value === 'string' &&
+    value.toLowerCase().includes(lowerCaseSearchTerm)
+
+  recordKeys.forEach(({ className: entity, recordKey }) => {
+    if (blackListEntities.includes(entity)) return
+
+    Object.entries(restoreRecord(recordKey)).forEach(([_id, item]) => {
+      if (Object.values(item).some(matchesSearchTerm)) {
+        results.push({
+          _id,
```

Comment:
* [ ] FYI We may end up with multiple results sharing the same `_id`, e.g., after copy-pasting to add a new local-storage data class. Not sure which implications this will have later on.